checkers/testdata/deferAtTheEnd/positive_tests.go

```diff
@@ -0,0 +1,49 @@
+package checker_test
+
+func foo1() {
+	/*! defer foo2() is placed just before return */
+	defer foo2()
+	return
+}
+
+func foo2() int {
+	/*! defer func(){}() is placed just before return */
+	defer func() {}()
+	return 0
+}
+
+func foo3() {
+	/*! defer func(){}() is placed just before return */
+	defer func() {}()
+}
+
+func foo4() {
+	/*! defer func(){…}() is placed just before return */
+	defer func() {
+		/*! defer foo1() is placed just before return */
+		defer foo1()
+		return
+	}()
+}
+
+func foo5() {
+	func() {
+		/*! defer foo1() is placed just before return */
+		defer foo1()
+		return
+	}()
+	foo1()
+	return
+}
+
+func foo6() {
+	/*! defer func(){…}() is placed just before return */
+	defer func() {
+		for {
+			/*! defer foo1() is placed just before return */
+			defer foo1()
+			return
+		}
+	}()
+	return
+}
```

Comment:
Looks good, I'm curious, will it handle `defer` inside `if`? :)
Even if no - it can be marked as a TODO comment.

Reply:
I'll add a test